### DISCUSSION TITLE
Highlighting dynamically generated content

### DISCUSSION THREAD

**Discussion by MixusMinimax** [MurhafSousli/ngx-highlightjs#217] [Q&A] [answered]
Hey, my website uses content delivery. I convert markdown text to html using marked, and store the result in innerHtml. Is there a way to highlight code segments contained in innerHtml using this?

**Comment by MurhafSousli** [comment, marked as answer]
The `[highlight]` directive will not be helpful in this use case, however, you can take advantage of the service call to the library

Try `highlightJS.highlightElement(contentWrapperEl).subrscribe` or `highlightJS.highlightAll().subscribe()`

https://github.com/MurhafSousli/ngx-highlightjs/blob/b4b7ba2d37cf8969e1dbc0958de4b3689d1d0e70/projects/ngx-highlightjs/src/lib/highlight.service.ts#L64-L74

https://github.com/MurhafSousli/ngx-highlightjs/blob/b4b7ba2d37cf8969e1dbc0958de4b3689d1d0e70/projects/ngx-highlightjs/src/lib/highlight.service.ts#L76-L84

**Reply by MixusMinimax** [reply]
Thanks! I have switched to `ngx-markdown` now, as it does everything for me anyway, luckily there is no reason to use marked anymore, as it messes up my dependencies anyway. This is still a nice project though, and I'm sure I will find a use for it in the future!

**Reply by MurhafSousli** [reply]
I've seen `ngx-markdown`, looks good! I am working on something similar which read MD content using marked and highlight.js but not as library.. more like a storybook app
